lib/Mojolicious/Plugin/OpenAPI/Guides/Security.pod

```diff
@@ -0,0 +1,110 @@
+=head1 NAME
+
+Mojolicious::Plugin::OpenAPI::Guides::Security - How to secure your API
+
+=head1 OVERVIEW
+
+This guide will give you an introduction on how to use the security features
+in OpenAPI, together with L<Mojolicious::Plugin::OpenAPI>.
+
+Note that this is currently EXPERIMENTAL! Please let me know if you have any
+feedback. See
+L<https://github.com/jhthorsen/mojolicious-plugin-openapi/pull/40> for a
+complete discussion.
+
+=head1 TUTORIAL
+
+=head2 Specification
+
+Here is an example specification that use
+L<securityDefinitions|http://swagger.io/specification/#securityDefinitionsObject>
+and L<security|http://swagger.io/specification/#securityRequirementObject> from
+the OpenAPI spec:
+
+  {
+    "swagger": "2.0",
+    "info": { "version": "0.8", "title": "Super secure" },
+    "schemes": [ "https" ],
+    "basePath": "/api",
+    "securityDefinitions": {
+      "dummy": {
+        "type": "apiKey",
+        "name": "Authorization",
+        "in": "header",
+        "description": "dummy"
+      }
+    },
+    "paths": {
+      "/protected": {
+        "post": {
+          "x-mojo-to": "super#secret_resource",
+          "security": [{"dummy": []}],
+          "parameters": [
+            { "in": "body", "name": "body", "schema": { "type": "object" } }
+          ],
+          "responses": {
+            "200": {"description": "Echo response", "schema": { "type": "object" }},
+            "401": {"description": "Sorry mate", "schema": { "type": "array" }}
+          }
+        }
+      }
+    }
+  }
+
+=head2 Application
+
+The specification above can be dispatched to handlers inside your
+L<Mojolicious> application. The do so, add the "security" key when loading the
+plugin, and reference the "securityDefinitions" name inside that to a callback.
+In this example, we have the "dummy" security handler:
+
+  package Myapp;
+  use Mojo::Base "Mojolicious";
+
+  sub startup {
+    my $app = shift;
+
+    $app->plugin(OpenAPI => {
+      url      => "data://main/sec.json",
+      security => {
+        dummy => sub {
+          my ($c, $definition, $scopes, $cb) = @_;
+          return $cb->(1) if $c->req->headers->authorization;
+          return $cb->(0);
```

Comment:
How about reversing the code, so we could do something like the code below? That way, we could use the first error, merge them or pass the errors on to some other handler at the end.

```
$c->$cb({error => "You shall not pass!"});
```